projects/RabbitMQ.Client/client/TaskExtensions.cs

```diff
@@ -59,21 +59,29 @@ public static Task TimeoutAfter(this Task task, TimeSpan timeout)
 
             return DoTimeoutAfter(task, timeout);
 
+            // https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#using-a-timeout
             static async Task DoTimeoutAfter(Task task, TimeSpan timeout)
             {
-                if (task == await Task.WhenAny(task, Task.Delay(timeout)).ConfigureAwait(false))
+                using (var cts = new CancellationTokenSource())
                 {
-                    await task.ConfigureAwait(false);
-                }
-                else
-                {
-                    Task supressErrorTask = task.ContinueWith(
-                        continuationAction: continuation,
-                        state: null,
-                        cancellationToken: CancellationToken.None,
-                        continuationOptions: s_tco,
-                        scheduler: TaskScheduler.Default);
-                    throw new TimeoutException();
+                    Task delayTask = Task.Delay(timeout, cts.Token);
+                    Task resultTask = await Task.WhenAny(task, delayTask).ConfigureAwait(false);
+                    if (resultTask == delayTask)
+                    {
+                        Task supressErrorTask = task.ContinueWith(
+                            continuationAction: continuation,
+                            state: null,
+                            cancellationToken: CancellationToken.None,
+                            continuationOptions: s_tco,
+                            scheduler: TaskScheduler.Default);
```

Comment:
Are you trying to suppress  the Task unobserved exception?

Reply:
@stebet originally wrote this code, so he can chime in. But yes, my understanding of the code is to suppress that exception.

Reply:
If it's not the correct way to do it, by all means suggest something else!

Reply:
Relevant discussion: https://github.com/dotnet/runtime/issues/23878

Reply:
I will be merging this PR as soon as CI completes, but I have the following issue open to continue review of `TaskExtensions` - https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1425